tests/InferTypesFuzz.elm

```diff
@@ -0,0 +1,293 @@
+intToIntFunctionExpr =
+    let
+        combine argument intPart =
+            Canonical.lambda argument <|
+                located <|
+                    Canonical.Plus
+                        (located <| Canonical.Argument argument)
+                        intPart
+
+        intSubExpr =
+            Type.Int |> exprGenerator
+    in
+    Random.map2 combine
+        -- TODO: Later we will need something better to avoid shadowing.
+        randomVarName
+        intSubExpr
+        |> Random.map located
+
+
+randomVarName : Generator VarName
+randomVarName =
+    let
+        starters =
+            "abcdefghijklnmoqprstuvwxyz"
+
+        others =
+            "ABCDEFGHIJKLMNOQPRSTUVQXYZ_0123456789"
+
+        all =
+            starters ++ others
+
+        charFrom string =
+            string
+                |> String.toList
+                |> List.map Random.constant
+                |> Random.choices (Random.constant 'a')
+
+        firstChar =
+            charFrom starters
+
+        rest =
+            Random.list 5 <| charFrom all
+    in
+    Random.map2 (::) firstChar rest
+        |> Random.map String.fromList
+        |> Random.map VarName
+
+
+dumpType : Type -> String
+dumpType type_ =
+    type_
+        |> Type.toString Type.emptyState
+        |> Tuple.first
+
+
+exprShrinker : Shrinker Canonical.LocatedExpr
+exprShrinker expr =
+    case
+        expr |> Located.unwrap
+    of
+        Canonical.Literal lit ->
+            lit
+                |> shrinkLiteral
+                |> Shrink.map Canonical.Literal
+                |> Shrink.map located
+
+        Canonical.Plus left right ->
+            expr
+                |> Debug.log "shrinking (+)"
+                |> shrinkPlus left right
+
+        Canonical.List elements ->
+            elements
+                |> shrinkNonEmptyList exprShrinker
+                |> Shrink.map Canonical.List
+                |> Shrink.map located
+
+        Canonical.If { then_, else_ } ->
+            [ then_, else_ ]
+                |> List.map shrinkTo
+                |> concatShrink
+                -- Lesser hack.
+                |> shrink expr
+
+        Canonical.Lambda { argument, body } ->
+            expr
+                |> shrinkLambda argument body
+
+        _ ->
+            expr |> Shrink.noShrink
+
+
+shrinkLiteral : Shrinker Literal
+shrinkLiteral lit =
+    case lit of
+        Int i ->
+            i |> Shrink.int |> Shrink.map Int
+
+        Float f ->
+            f |> Shrink.float |> Shrink.map Float
+
+        Char c ->
+            c |> Shrink.char |> Shrink.map Char
+
+        Bool b ->
+            b |> Shrink.bool |> Shrink.map Bool
+
+        String s ->
+            s |> Shrink.string |> Shrink.map String
+
+
+shrinkPlus : Canonical.LocatedExpr -> Canonical.LocatedExpr -> Shrinker Canonical.LocatedExpr
+shrinkPlus left right _ =
+    -- TODO: The Shrink docs were misleading here. Consider reporting an issue.
```

Comment:
Care to elaborate more in the comment (or report the issue right away) while you still have the context in your head? This comment as it is now, doesn't give me enough information to understand what is the problem

Reply:
I'm still doing some work on the PR. The comment is no longer present locally.

The issue is reported at https://github.com/elm-explorations/test/issues/101.